internal/v4/stats.go

```diff
@@ -110,3 +110,12 @@ func (h *Handler) serveStatsCounter(_ http.Header, r *http.Request) (interface{}
 
 	return items, nil
 }
+
+// StatsEnabled reports whether statistics should be gathered for
+// the given HTTP request.
+func StatsEnabled(req *http.Request) bool {
+	// It's fine to parse the form more than once, and it avoids
```

Comment:
Not necessarily in the legacy case. I don't mind leaving it there - it's only a few ns.